File: ekobloc/the_block/doc_events/sales_invoice.py
import frappe

from erpnext.accounts.utils import (
	get_balance_on
)

@frappe.whitelist()
def get_outstanding_amount(company,date,party):
    customer_outstanding_amount = get_balance_on(
	    party_type="Customer", party=party, date=date, company=company
				)
    return customer_outstanding_amount


def get_outstanding_amount_validation(self,method):
    if self.customer_outstanding_amount > 0:
        frappe.throw("You are not eligible To Book Sales Invoice ")
    
    if self.customer_outstanding_amount + self.net_total > 0:
        frappe.throw("Your Amount is less Please Payment First")    
            

File: ekobloc/public/js/sales_invoice.js
// frappe.ui.form.on('Payment Entry', {
//     customer:function(frm){
//         frappe.call({
//             method:"ekobloc.the_block.doc_events.sales_invoice.get_outstanding_amount",
//             args:{
//                 company:frm.doc.company,
//                 date: frm.doc.posting_date,
//                 party: frm.doc.customer

//             },
//             callback: function(r){
//                 if(r.message){
//                     frm.set_value("customer_outstanding_amount",r.message)
//                 }
//             }
//         })
//     }
// })

frappe.ui.form.on('Sales Invoice', {
    customer: function(frm) {
        if (frm.doc.posting_date && frm.doc.company && frm.doc.customer) {
       

        frappe.call({
            method: "ekobloc.the_block.doc_events.sales_invoice.get_outstanding_amount",
            args: {
                company: frm.doc.company,
                date: frm.doc.posting_date,
                party: frm.doc.customer
            },
            callback: function(r) {
                if (r.message) {
                    frm.set_value("customer_outstanding_amount", r.message);
                }
            }
        })
    }
    }
})
